src/replication/ng/scheduler/schedule.go

```diff
@@ -0,0 +1,92 @@
+package schedule
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/goharbor/harbor/src/common/job"
+	"github.com/goharbor/harbor/src/common/job/models"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/jobservice/opm"
+	"github.com/goharbor/harbor/src/replication/ng/model"
+)
+
+// DefaultReplicator provides a default implement for Replicator
+type DefaultReplicator struct {
+	client job.Client
+}
+
+// NewDefaultReplicator returns an instance of DefaultReplicator
+func NewDefaultReplicator(client job.Client) *DefaultReplicator {
+	return &DefaultReplicator{
+		client: client,
+	}
+}
+
+// Scheduler to schedule the tasks to transfer resource data
+type Scheduler interface {
+	// Schedule tasks
+	Schedule(srcResources []*model.Resource, destResources []*model.Resource) ([]*model.Task, error)
+	StopTransfer(jobID string) error
+}
+
+// Schedule the task to transfer resouce data
+func (d *DefaultReplicator) Schedule(srcResources []*model.Resource, destResources []*model.Resource) ([]*model.Task, error) {
+	var tasks []*model.Task
+	for _, destResource := range destResources {
```

Comment:
The srcResource:dstResource is 1:1, not m:n, that means srcResources[0] will be replicated to dstResources[0]

Reply:
what if the two parameters got different length? 
should I throw a exception and quit? 
or should I just transfer the resources as much as what the shorter length is?